src/trie.rs

```diff
@@ -362,20 +440,34 @@ where
                 let prefix = &extension.prefix;
                 let match_len = partial.common_prefix(prefix);
                 if match_len == prefix.len() {
-                    self.get_at(extension.node.clone(), &partial.offset(match_len))
+                    self.get_at(extension.node.clone(), path, path_index + match_len)
                 } else {
                     Ok(None)
                 }
             }
             Node::Hash(hash_node) => {
-                let borrow_hash_node = hash_node.borrow();
-                let n = self.recover_from_db(&borrow_hash_node.hash)?;
-                self.get_at(n, partial)
+                let node_hash = hash_node.borrow().hash.clone();
+                let node = self.recover_from_db(&node_hash)?.ok_or_else(|| {
```

Comment:
How does this ever trigger the `else` clause, if anything other than `Ok` will be short-circuited by the `?` before `ok_or_else`?

Haven't run this, and I might be mis-reading.